dotnet/src/Microsoft.Agents.AI.Workflows/ChatProtocolExecutor.cs

```diff
@@ -8,23 +8,46 @@
 
 namespace Microsoft.Agents.AI.Workflows;
 
-internal class ChatProtocolExecutorOptions
+/// <summary>
+/// Provides configuration options for <see cref="ChatProtocolExecutor"/>.
+/// </summary>
+public class ChatProtocolExecutorOptions
 {
+    /// <summary>
+    /// Gets or sets the chat role to use when converting string messages to <see cref="ChatMessage"/> instances.
+    /// If set, the executor will accept string messages and convert them to chat messages with this role.
+    /// </summary>
     public ChatRole? StringMessageChatRole { get; set; }
 }
 
-// TODO: Make this a public type (in a later PR; todo: make an issue)
-internal abstract class ChatProtocolExecutor : StatefulExecutor<List<ChatMessage>>
+/// <summary>
+/// Provides a base class for executors that implement the Agent Workflow Chat Protocol.
+/// This executor maintains a list of chat messages and processes them when a turn is taken.
+/// </summary>
+/// <remarks>
+/// The Chat Protocol is defined as supporting <see cref="List{ChatMessage}"/>, <see cref="ChatMessage"/>,
+/// arrays and collections of chat messages, and <see cref="TurnToken"/> as input. When a turn token is received,
+/// the accumulated messages are processed via the <see cref="TakeTurnAsync(List{ChatMessage}, IWorkflowContext, bool?, CancellationToken)"/>
+/// method, which derived classes must implement.
+/// </remarks>
+public abstract class ChatProtocolExecutor : StatefulExecutor<List<ChatMessage>>
 {
     private readonly static Func<List<ChatMessage>> s_initFunction = () => [];
     private readonly ChatRole? _stringMessageChatRole;
 
-    internal ChatProtocolExecutor(string id, ChatProtocolExecutorOptions? options = null, bool declareCrossRunShareable = false)
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ChatProtocolExecutor"/> class.
+    /// </summary>
+    /// <param name="id">The unique identifier for this executor instance. Cannot be null or empty.</param>
+    /// <param name="options">Optional configuration settings for the executor. If null, default options are used.</param>
+    /// <param name="declareCrossRunShareable">true to declare that the executor's state can be shared across multiple runs; otherwise, false.</param>
```

Comment:
The point of `crossRunShareable` is not that the Executor's state can be shared, but that the executor instance can be shared.